wiki: added custom auth description (#1266)
artipie/artipie: .wiki/Configuration-Credentials.md

Artipie acts as Keycloak client application and should be configured in Keycloak with following settings :
* Client authentication: On
* Authorization: On
* Authentication flow: `Direct access grants`

## Custom authentication

Artipie allows implementing and using custom authentication (credentials type). To be more precise,
you can choose any way to authenticate users you like, implement it in Java and use in Artipie. To do
so:

> - add [`http` module](https://github.com/artipie/http) to your project dependencies 

Reviewer: ```suggestion
- Add [`http` module](https://github.com/artipie/http) to your project dependencies.
```

- Implement [Authentication](https://github.com/artipie/http/blob/master/src/main/java/com/artipie/http/auth/Authentication.java) interface 
to verify and obtain users the way you need. Note, that authenticated users are cached on the upper level,
so, there is no need to worry about caching in your implementation.
- Implement [factory](https://github.com/artipie/http/blob/master/src/main/java/com/artipie/http/auth/AuthFactory.java) for new authentication
and add [factory annotation](https://github.com/artipie/http/blob/master/src/main/java/com/artipie/http/auth/ArtipieAuthFactory.java) 
with authentication type name. As an example, check [KeycloakFactory](https://github.com/artipie/artipie/blob/master/src/main/java/com/artipie/auth/AuthFromKeycloakFactory.java).

> - Add your package to Artipie class-path and to `AUTH_FACTORY_SCAN_PACKAGES` environment variable

Reviewer 2: I would prefer to add detailed information about "How to add to Artipie classpath".
Can we say "Put your jar to lib folder of Artipie" or we have some other standard way for it?

One of ideas for feature enhancements is to create special folder for external jars for example it can be named 'ext' and Artipie should load from it all jars

Author: good suggestion, but simply say "Put your jar to lib folder of Artipie" is not enough as we do not know how the user want to run Artipie - as docker image or jar or from IDE. I guest the best way here will be to create some separate page describing all the possibilities and add links in custom storages, policy docs and here...

> - Specify your authentication type in credentials list. Note, that Artipie try to authenticate 

Reviewer: ```suggestion
- Specify your authentication type in the credentials list. Note, that Artipie tries to authenticate 
```

user via each listed type. So, if you are not actually using any auth type it would be better 

> to remove it from settings.

Reviewer: ```suggestion
to remove it from the settings.
```
